probe-rs/src/probe.rs

```diff
@@ -252,13 +252,12 @@ impl Probe {
     ///
     /// If this doesn't work, you might want to try [`Probe::attach_under_reset`]
     pub fn attach(
-        mut self,
+        self,
         target: impl Into<TargetSelector>,
         permissions: Permissions,
     ) -> Result<Session, Error> {
-        self.attached = true;
-
-        Session::new(self, target.into(), AttachMethod::Normal, permissions)
+        let session = Session::new(self, target.into(), AttachMethod::Normal, permissions)?;
```

Comment:
Any reason this is two statements now? Pretty sure clippy does not like this.

Reply:
Oops! That was a leftover change from something else, fixed now :)